### PR TITLE
Add `grpcio` metrics exporter

### PR DESCRIPTION
**Pull request by jtescher**
## Changes

This adds missing `grpcio` support for exporting metrics in OTLP.

Instead of porting the metrics transformations to the unsupported `protobuf` v2.x impl (v3 is not supported by grpcio), this switches grpcio to use prost as well, this simplifies and unifies the proto transformations to the respective `tonic` and `grpcio` ecosystems.

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-proto/tests/grpc_build.rs [hunk 1]
```diff
@@ -97,6 +89,8 @@ fn ensure_files_are_same(
     after_build: HashMap<String, String>,
     target_dir: &'static str,
 ) {
+    dbg!(&before_build.keys());
+    dbg!(&after_build.keys());
```
Comment (TommyCpp) [inline opentelemetry-proto/tests/grpc_build.rs:93]: Do we still need those?

File: opentelemetry-otlp/src/metric.rs [hunk 2]
```diff
@@ -80,10 +91,23 @@ impl MetricsExporterBuilder {
...
             MetricsExporterBuilder::Http(builder) => {
                 builder.build_metrics_exporter(aggregation_selector, temporality_selector)
             }
+
+            #[cfg(not(any(feature = "http-proto", feature = "grpc-sys", feature = "grpc-tonic")))]
+            MetricsExporterBuilder::Unconfigured => {
+                drop(temporality_selector);
+                drop(aggregation_selector);
+                Err(opentelemetry_api::metrics::MetricsError::Other(
+                    "no configured span exporter".into(),
```
Comment (TommyCpp) [inline opentelemetry-otlp/src/metric.rs:108]:
```suggestion
                    "no configured metrics exporter, enable `http-proto`, `grpc-sys` or `grpc-tonic` feature to configure a metrics exporter".into(),
```

**Review by TommyCpp — APPROVED** [review]
Mostly LGTM. Small nits

File: opentelemetry-proto/tests/grpc_build.rs (hunk 1 above)
Comment (jtescher) [reply opentelemetry-proto/tests/grpc_build.rs:93]: ah no, good catch

**Pull request merged**